mono/metadata/appdomain.c

```diff
@@ -546,7 +546,8 @@ mono_domain_create_appdomain_internal (char *friendly_name, MonoAppDomainSetup *
 	data->domain = ad;
 	data->friendly_name = g_strdup (friendly_name);
 
-	mono_profiler_appdomain_name (data, data->friendly_name);
+	if (mono_profiler_events & MONO_PROFILE_APPDOMAIN_EVENTS)
```

Comment:
I really don't like this change.

The profiler events are checked anyways on the calls. Plus we're not hinting the compiler that it's a non taken branch.

We should use something like this:

```
#define GEN_EVT2(EVENT, FUNC, ARG0, ARG1)	do { \
	if (G_UNLIKELY (mono_profiler_events & MONO_PROFILE_ ## EVENT))	\
		mono_profiler_ ## FUNC (ARG0, ARG1);	\
} while (0)

#define RAISE_APPDOMAIN_NAME(ARG0, ARG1) GEN_EVT2( APPDOMAIN_EVENTS, appdomain_name, ARG0, ARG1);

```